src/main/java/com/bringframework/context/AnnotationConfigApplicationContext.java

```diff
@@ -11,23 +14,23 @@
 import lombok.extern.slf4j.Slf4j;
 
 /**
- * Implementation of {@code ApplicationContext} that uses annotation configuration to search and
+ * Implementation of {@link ApplicationContext} that uses annotation configuration to search and
  * create beans.
  */
 @Slf4j
 public class AnnotationConfigApplicationContext implements ApplicationContext {
 
-  private Map<String, Object> beans;
+  private final Map<String, Object> beans;
 
-  public AnnotationConfigApplicationContext() {
+  public AnnotationConfigApplicationContext(String packageName) {
+    validatePackageName(packageName);
     log.trace("Application context is collecting...");
     BeanDefinitionRegistry beanDefinitionRegistry = new DefaultBeanDefinitionRegistry();
-    BeanDefinitionReader beanDefinitionReader = new DefaultBeanDefinitionReader(
-        beanDefinitionRegistry);
+    new DefaultBeanDefinitionReader(beanDefinitionRegistry).registerBeans(packageName);
```

Comment:
it is already `BeanDefinitionScanner`, please update with latest changes from dev